src/Microsoft.IdentityModel.Tokens/Base64UrlEncoder.cs

```diff
@@ -207,37 +236,19 @@ internal static unsafe byte[] UnsafeDecode(ReadOnlySpan<char> strSpan)
                     arrayPoolChars = ArrayPool<char>.Shared.Rent(decodedLength);
                 charsSpan = charsSpan.Slice(0, decodedLength);
 
-                source.CopyTo(charsSpan);
-                if (source.Length < charsSpan.Length)
-                {
-                    charsSpan[source.Length] = base64PadCharacter;
-                    if (source.Length + 1 < charsSpan.Length)
-                    {
-                        charsSpan[source.Length + 1] = base64PadCharacter;
-                    }
-                }
-
-                if (needReplace)
-                {
-                    Span<char> remaining = charsSpan;
-                    int pos;
-                    while ((pos = remaining.IndexOfAny(base64UrlCharacter62, base64UrlCharacter63)) >= 0)
-                    {
```

Comment:
This is potentially non-optimal. If I'm understanding it correctly, in an extreme case scenario it could end up being O(n^2) (?) Wouldn't it make sense to simply iterate over it once with a simple for loop and replace in place?